api/src/main/java/marquez/db/JobVersionDao.java

```diff
@@ -136,16 +163,22 @@ default JobVersionBag createJobVersionOnComplete(
 
     createRunDao().updateJobVersion(runUuid, jobVersion.getUuid());
 
-    jobDao.updateVersion(jobRow.getUuid(), transitionedAt, jobVersion.getUuid());
+    // Only set current version if completed
```

Comment:
Do we only want to set the latest version on complete? We also create new versions when a job run is [marked as _done_](bourdains).

Reply:
Yeah. The latest dataset version is the latest version that was successful. This makes sense in the context of a lineage graph as the graph can still be constructed w/ the latest run returned is the actual latest run which could be in a failed state.

Reply:
Sounds good. When we introduce the JobVersion API, we'll want to document the logic for how the current version is set.